components/core/src/clp_s/JsonParser.cpp

```diff
@@ -520,6 +521,317 @@ bool JsonParser::parse() {
     return true;
 }
 
+NodeType get_archive_node_type(
+        clp::ffi::SchemaTreeNode const& node,
+        std::pair<clp::ffi::SchemaTreeNode::id_t, std::optional<clp::ffi::Value>> p
+) {
+    auto const node_type = node.get_type();
+    // figure out what type the node is in archive node type
+    NodeType archiveNodeType;
+    switch (node_type) {
+        case clp::ffi::SchemaTreeNode::Type::Int:
+            archiveNodeType = NodeType::Integer;
+            break;
+        case clp::ffi::SchemaTreeNode::Type::Float:
+            archiveNodeType = NodeType::Float;
+            break;
+        case clp::ffi::SchemaTreeNode::Type::Bool:
+            archiveNodeType = NodeType::Boolean;
+            break;
+        case clp::ffi::SchemaTreeNode::Type::UnstructuredArray:
+            archiveNodeType = NodeType::UnstructuredArray;
+            break;
+        case clp::ffi::SchemaTreeNode::Type::Str:
+            // std::cerr << "In str\n";
+            if (p.second.value().is<std::string>()) {
+                // maybe special case for date string
+                archiveNodeType = NodeType::VarString;
+            } else {
+                archiveNodeType = NodeType::ClpString;
+            }
+            break;
+        case clp::ffi::SchemaTreeNode::Type::Obj:
+            // std::cerr << "In obj\n";
+            if (p.second.has_value()) {
+                if (p.second.value().is_null()) {
+                    // std::cout << "Found Null\n";
+                    archiveNodeType = NodeType::NullValue;
+                } else {
+                    archiveNodeType = NodeType::Object;
+                }
+            } else {
+                archiveNodeType = NodeType::Object;
+            }
+            break;
+        default:
+            archiveNodeType = NodeType::Unknown;
+            break;
+    }
+    return archiveNodeType;
+}
+
+//
+int JsonParser::get_archive_node_id(
+        std::map<std::tuple<int, NodeType>, int>& cache,
+        int irNodeID,
+        NodeType archiveNodeType,
+        clp::ffi::SchemaTree const& irTree
+) {
+    std::tuple<int, NodeType> key(irNodeID, archiveNodeType);
+    if (cache.find(key) != cache.end()) {
+        return cache[key];
+    }
+    auto& currNode = irTree.get_node(irNodeID);
+    int parent_node_id;
+    // Found the root
+    if (currNode.get_parent_id() == 0) {
+        parent_node_id = 0;
+    } else {
+        parent_node_id
+                = get_archive_node_id(cache, currNode.get_parent_id(), NodeType::Object, irTree);
+    }
+    std::string nodeKey
+            = clp::ffi::validate_and_escape_utf8_string(currNode.get_key_name()).value();
```

Comment:
Is it safe to call .value() on the result of validate_and_escape_utf8_string directly here?

What happens when validation fails?

Reply:
I added a check for if this .has_value(). If it doesn't have value should we error and return/exit or should be just insert a empty string as the key?